tools/ctu-analysis/ctu-build.py

```diff
@@ -133,11 +133,9 @@ def map_functions(params):
     command, sources, directory, clang_path, ctuindir = params
     args = get_command_arguments(command)
     dir_command = ['cd', directory]
-    funcmap_command = [os.path.join(clang_path, 'clang-func-mapping')]
-    funcmap_command.append('--ctu-dir')
-    funcmap_command.append(os.path.abspath(ctuindir))
+    funcmap_command = [os.path.join(clang_path, 'clang-func-mapping'),
+                       '--ctu-dir', os.path.abspath(ctuindir), '--']
```

Comment:
Have you tried this out? Originally, sources are separated by '--' from the arguments. Now, '--' comes first and then sources and args together.